app/src/main/java/uk/ac/ebi/atlas/experimentpage/JsonMarkerGenesController.java

```diff
@@ -43,4 +43,23 @@ public String getCellTypeMarkerGenes(@PathVariable String experimentAccession,
                 markerGeneService.getCellTypeMarkerGeneProfile(experimentAccession, ImmutableSet.copyOf(organismPart))
         ));
     }
+
+    @GetMapping(value = "/json/experiments/{experimentAccession}/marker-genes-heatmap/cell-types",
+            produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
+    public String getCellTypeMarkerGenesHeatmap(@PathVariable String experimentAccession,
+                                                @RequestParam String cellGroupType,
+                                                @RequestParam Set<String> cellType) {
+        return GSON.toJson(highchartsHeatmapAdapter.getMarkerGeneHeatmapDataSortedLexicographically(
+                markerGeneService.getCellTypeMarkerGeneHeatmap(experimentAccession, cellGroupType, ImmutableSet.copyOf(cellType))
+        ));
+    }
+
+    @GetMapping(value = "/json/experiments/{experimentAccession}/marker-genes-heatmap/cellTypeGroups",
+            produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
+    public String getCellTypeGroupsMarkerGenesHeatmap(@PathVariable String experimentAccession,
+                                                      @RequestParam String cellGroupType) {
+        return GSON.toJson(
+                markerGeneService.getCellTypesWithMarkerGenes(experimentAccession, cellGroupType)
+        );
+    }
```

Comment:
Fetch cell types for the heatmap by giving input as a cell group type(ontology or authors labels) & experiment accession
endpoint -` /json/experiments/{experimentAccession}/marker-genes-heatmap/cell-types`
method- `getCellTypesForMarkerGenesHeatmap`

Probably you missed our summary of the meeting points :)

Reply:
For the endpoint `/json/experiments/{experimentAccession}/marker-genes-heatmap/cell-types`, which retrieve the heatmap data, we need to pass both `cellTypeGroup` and `cellTypes`.

I renamed the endpoint and method to clarify we are retrieving `heatmapData`, as `getCellTypeMarkerGenesHeatmapData` and `getCellTypeMarkerGeneHeatmapData` respetively.